Source/Core/DolphinTool/ConvertCommand.cpp

```diff
@@ -0,0 +1,276 @@
+    {
+      std::cerr << "Error: Compression format and level must be set for WIA or RVZ" << std::endl;
+      return 1;
+    }
+
+    const std::pair<int, int> range = DiscIO::GetAllowedCompressionLevels(compression_o.value());
+    if (compression_level_o.value() < range.first || compression_level_o.value() > range.second)
+    {
+      std::cerr << "Error: Compression not in acceptable range for the format" << std::endl;
+      return 1;
+    }
+  }
+
+  // Open the blob reader
+  std::unique_ptr<DiscIO::BlobReader> blob_reader =
+      scrub ? DiscIO::ScrubbedBlob::Create(input_file_path) :
+              DiscIO::CreateBlobReader(input_file_path);
+  if (!blob_reader)
+  {
+    std::cerr << "Error: Unable to process disc image. If --scrub is enabled, try again without it."
+              << std::endl;
+    return 1;
+  }
+
+  // Perform the conversion
+  const auto NOOP_STATUS_CALLBACK = [](const std::string& text, float percent) { return true; };
+
+  std::future<bool> success;
+
+  switch (format)
+  {
+  case DiscIO::BlobType::PLAIN:
+    success = std::async(std::launch::async, [&] {
+      const bool good = DiscIO::ConvertToPlain(blob_reader.get(), input_file_path, output_file_path,
+                                               NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  case DiscIO::BlobType::GCZ:
+    success = std::async(std::launch::async, [&] {
+      const bool good =
+          DiscIO::ConvertToGCZ(blob_reader.get(), input_file_path, output_file_path,
+                               volume->GetVolumeType() == DiscIO::Platform::WiiDisc ? 1 : 0,
+                               block_size_o.value(), NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  case DiscIO::BlobType::WIA:
+  case DiscIO::BlobType::RVZ:
+    success = std::async(std::launch::async, [&] {
+      const bool good = DiscIO::ConvertToWIAOrRVZ(
+          blob_reader.get(), input_file_path, output_file_path, format == DiscIO::BlobType::RVZ,
+          compression_o.value(), compression_level_o.value(), block_size_o.value(),
+          NOOP_STATUS_CALLBACK);
+      return good;
+    });
+    break;
+
+  default:
+    ASSERT(false);
+    break;
+  }
+
+  if (!success.get())
+  {
+    std::cerr << "Error: Conversion failed" << std::endl;
+    return 1;
+  }
+
+  std::cout << "Success" << std::endl;
+  return 0;
+}
+
+bool ConvertCommand::CheckBlockSize(const int block_size, const u64 file_size,
+                                    const DiscIO::BlobType format)
+{
+  if (format == DiscIO::BlobType::GCZ)
+  {
+    // Refer to ConvertDialog.cpp
+    if (!(file_size % block_size == 0 &&
+          file_size % (block_size * ConvertCommand::GCZ_BLOCKS_PER_BUFFER) != 0))
+    {
+      std::cerr << "Error: Invalid block size for GCZ. File size must be an integer multiple of "
+                   "the block size "
+                   "and must not be an integer multiple of the block size multiplied by 32.";
```

Comment:
If the user *explicitly* enters something which old versions of Dolphin have trouble with, I think it makes more sense to just show a warning instead of aborting. The command-line interface is more of a power user interface than DolphinQt after all.